### DISCUSSION TITLE
How to dynamically update a second dropdown in FlutterFlow based on the first dropdown selection?

### DISCUSSION THREAD

**Discussion by Rajkretoss** [community/community#147242] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

I am working on a FlutterFlow application where I have two dropdowns with the following requirements:

The first dropdown allows users to select a group, and the selected group's ID should be stored in an App Variable.
This App Variable needs to be used in an API call to populate the second dropdown dynamically.
Additionally:

If the API response for the second dropdown is an empty array, the second dropdown should be disabled.
If no selection is made in the first dropdown, the second dropdown should also be disabled.

I attempted to store the selected group's ID in an App Variable, but I am unsure how to bind this variable to the API call dynamically.

I tried using FlutterFlow actions to manage the second dropdown’s state, but it did not work as expected.

Expected Outcome:
The second dropdown updates dynamically based on the first dropdown's selection and the API response.
It remains disabled if the API response is empty or no value is selected in the first dropdown.

**Comment by xDweeb** [comment, marked as answer]
Ensure the App Variable is updated on the first dropdown selection and bind it to the API call for the second dropdown. Use conditions to disable the second dropdown if the API response is empty or no selection is made.

**Reply by Rajkretoss** [reply]
It's working. Thanks @xDweeb.
